src/sentry/static/sentry/app/views/projects/redirectSentry9Project.jsx

```diff
@@ -0,0 +1,157 @@
+import React from 'react';
+import {withRouter} from 'react-router';
+import PropTypes from 'prop-types';
+import _ from 'lodash';
+
+import {t} from 'app/locale';
+import withApi from 'app/utils/withApi';
+import LoadingError from 'app/components/loadingError';
+
+// TODO: This is react-router v4 <Redirect to="path/" /> component to allow things
+//       to be declarative
+export class Redirect extends React.Component {
+  static propTypes = {
+    router: PropTypes.object.isRequired,
+    to: PropTypes.oneOfType([PropTypes.string, PropTypes.object]).isRequired,
+  };
+
+  componentDidMount() {
+    this.props.router.replace(this.props.to);
+  }
+
+  render() {
+    return null;
+  }
+}
+
+class ProjectDetailsInner extends React.Component {
+  static propTypes = {
+    api: PropTypes.object.isRequired,
+
+    orgId: PropTypes.string.isRequired,
+    projectId: PropTypes.string.isRequired,
+  };
+
+  state = {
+    loading: true,
+    error: null,
+    project: null,
+  };
+
+  componentDidMount() {
+    this.fetchData();
+  }
+
+  fetchData = async () => {
+    this.setState({
+      loading: true,
+      error: null,
+    });
+
+    const {orgId, projectId} = this.props;
+
+    try {
+      const projectRequest = this.props.api.requestPromise(
+        `/projects/${orgId}/${projectId}/`
+      );
+
+      const project = await projectRequest;
+      this.setState({
+        loading: false,
+        error: null,
+        project,
+      });
+    } catch (error) {
+      this.setState({
+        loading: false,
+        error,
+        project: null,
+      });
+    }
+  };
+
+  getProjectId = () => {
+    return _.get(this.state.project, 'id');
+  };
+
+  hasProjectId = () => {
+    const projectID = this.getProjectId(this.state.project);
+    return _.isString(projectID) && projectID.length > 0;
+  };
+
+  render() {
+    const childrenProps = {
+      ...this.state,
+      getProjectId: this.getProjectId,
```

Comment:
Hmm we generally just pass the evaluated func, it would keep it a bit simpler, components downstream can just expect a single `projectId` value that can either be `String|null`